inc/sainexthop.h

```diff
@@ -229,6 +229,28 @@ typedef enum _sai_next_hop_attr_t
      */
     SAI_NEXT_HOP_ATTR_DECREMENT_TTL,
 
+    /**
+     * @brief Encap tunnel-id for MPLS PUSH
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_TUNNEL
+     * @allownull true
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_MPLS
+     */
+    SAI_NEXT_HOP_ATTR_MPLS_ENCAP_TUNNEL_ID,
```

Comment:
is this behavior backward compatible? it looks now if the next hop is mpls, then user must include this new attribute? it seems like a non-backward compatible behavior.

Reply:
this is not mandatory attribute, it is proposed for PUSH operation to indicate TTL and QoS mode